### PR TITLE
Add verification API call

### PR DESCRIPTION
**Pull request by thomasspriggs**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by NlightNFotis — APPROVED** [review]

**Comment by NlightNFotis** [comment]
This looks good to me as it stands.

Should this remain as a draft, or should we get this out of draft and in `develop`?

**Comment by NlightNFotis** [comment]
The fix for the `cpplint` issue is to add these lines in `src/libcprover-cpp/module_dependencies.txt`:

```
assembler
goto-checker
pointer-analysis
```

File: src/libcprover-cpp/api.cpp [hunk 1]
```diff
@@ -105,3 +115,47 @@ void api_sessiont::load_model_from_files(const std::vector<std::string> &files)
   implementation->model = util_make_unique<goto_modelt>(initialize_goto_model(
     files, *implementation->message_handler, *implementation->options));
 }
+
+void api_sessiont::verify_model()
```
Comment (esteffin) [inline src/libcprover-cpp/api.cpp:119]: ⛏️ 
In my opinion this function is a bit too long and complex to be in an API.

Can it be rewritten moving the logics into `goto-checker` or somewhere else?

**Review by esteffin — APPROVED** [review]
Looks good to me except for a minor non-blocking comment.

File: src/libcprover-cpp/api.cpp (hunk 1 above)
Comment (thomasspriggs) [reply src/libcprover-cpp/api.cpp:119]: This reflects the complexity of the existing internal APIs. There isn't a whole lot we can do without refactoring the rest of the code base, for which would need buy-in to whatever we come up with from more code owners. If its any consolation, this is a lot easier to go through than `cbmc_parse_options.cpp`, which is the equivalent for the `cbmc` command line binary. A lot of what is in this function is also comments, explaining why the calls are needed in the order they are in.

**Review by peterschrammel — COMMENTED** [review]
I understand that this is a shim to get something off the ground. However, going forward,  I'd generally prefer that the API was developed incrementally in a way to replace functionality in cbmc_parse_options instead of duplicating the code... with all the advantages that come with it (having an API user from the beginning and thus more thoughtfulness about the choices made; full testing on all regression tests; no risk of divergence in duplicated code, etc)

**Pull request merged**